### DISCUSSION TITLE
asyncio.TimeoutError not working or just i dont know how to use it???

### DISCUSSION THREAD

**Discussion by SoftDevDanial** [Pycord-Development/pycord#491] [Q&A] [answered]
So i have a working music bot.
```
try:
  asyncio.run_coroutine_threadsafe(self.bot.wait_for("message",timeout=timeout, check = lambda m: m.content.startswith('*')),self.bot.loop)
  print("im Here 1")
except asyncio.TimeoutError:
  print("im Here 2 ")
  asyncio.run_coroutine_threadsafe(self.leave(ctx),self.bot.loop)
```

And I want to make the bot leave when nobody commands the bot to do anything after it finishes playing the final song. So according to the docs, i should use the bot.wait_for, add the command parameter in strings, add the timeout and do a check. But right now, i am not getting into the except `asyncio.TimeoutError` and i am not getting any timeout raised either. (i am using `asyncio.run_coroutine_threadsafe` because i am running this code in a sync function. Why? IDK, did it in an async function but there were problems stating i did not await the async functions inside it eventhough i already wrote await.) Anyway, anyone know whats wrong with this piece of code or do i need to add more of my code? Everything runs just fine. Is just that it get stuck here and it only prints "Im Here 1" but not "Im Here 2"

P/s: i did 5 seconds timeout, 30 seconds and even 120 seconds. If that makes any difference lol

**Comment by zeffo** [comment, marked as answer]
run_coroutine_threadsafe returns a Future. Exceptions raised in the coroutine will be propagated to the Future. In any case, you really shouldn't be using running coroutines like this.
